charts/tyk-gateway/templates/deployment.yaml

```diff
@@ -0,0 +1,169 @@
+apiVersion: apps/v1
+kind: {{ .Values.kind }}
+metadata:
+  name: {{ include "tyk-gateway.fullname" . }}
+  labels:
+    {{- include "tyk-gateway.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "tyk-gateway.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "tyk-gateway.selectorLabels" . | nindent 8 }}
+    spec:
+{{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+{{- end }}
+      initContainers:
+      - name: "setup-directories"
+        image: busybox:1.32
+        command: ['sh','-c','mkdir -p apps middleware policies && touch policies/policies.json']
+        workingDir: /mnt/tyk-gateway
+        volumeMounts:
+          - name: tyk-scratch
+            mountPath: /mnt/tyk-gateway
+          {{- if .Values.extraMounts }}
+          {{- range .Values.extraMounts }}
+          - name: {{ .Values.extraMounts.name }}
+            mountPath: {{ .Values.extraMounts.mountPath }}
```

Comment:
This will actually fail when rendering or deploying the chart. When using `range` context (parent) scope must be set using `$` to access/read `.Values`